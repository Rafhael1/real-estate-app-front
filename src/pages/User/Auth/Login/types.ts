export const ACTIONS = {
	LOGIN_REQUEST: 'LOGIN_REQUEST',
	LOGIN_SUCCESS: 'LOGIN_SUCCESS',
	LOGIN_ERROR: 'LOGIN_ERROR',
}

type UserType = {
	id: string,
	name: string,
    email: string,
    password: string
}[]

export interface IState {
    isLoading: boolean,
    hasError: boolean,
    user?: UserType,
	isUserLogged: boolean
}

export interface LoginRequest {
	type: typeof ACTIONS.LOGIN_REQUEST
}

export interface LoginSuccess {
	type: typeof ACTIONS.LOGIN_SUCCESS
}

export interface LoginError {
	type: typeof ACTIONS.LOGIN_ERROR
}

export type LoginDispatchTypes = LoginRequest | LoginSuccess | LoginError
