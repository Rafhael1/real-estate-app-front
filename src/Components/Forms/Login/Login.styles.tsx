import { makeStyles } from '@mui/styles';

const useStyles = makeStyles(() => ({
  dialogTitle: {
    fontWeight: 900
  },
  loginButton: {
    width: '200px'
  },
  signUpLink: {
    cursor: 'pointer',
    textDecoration: 'underline',
    alignItems: 'center',
    marginTop: '15px'
  }
}));

export default useStyles;
