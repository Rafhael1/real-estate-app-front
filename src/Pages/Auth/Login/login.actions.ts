import { createAsyncThunk } from '@reduxjs/toolkit';
import axios from '../../../Config/axios';

import { UserType } from './login.types';

export const login: any = createAsyncThunk(
  'login',
  async (values: UserType) => {
    try {
      const res = (await axios.post('/authentication/login', values)).records;
      if (res.authToken) {
        if (values.rememberMe) {
          localStorage.setItem('authToken', res.authToken);
        } else if (!values.rememberMe) {
          sessionStorage.setItem('authToken', res.authToken);
        }
      }
      return res;
    } catch (error) {
      return error;
    }
  }
);
