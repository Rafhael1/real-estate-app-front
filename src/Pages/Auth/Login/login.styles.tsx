import { makeStyles } from '@mui/styles';

const useStyles: any = makeStyles(() => ({}));

export default useStyles;
