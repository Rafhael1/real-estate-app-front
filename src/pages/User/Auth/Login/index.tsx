import React from "react"

import { Form, Field, reduxForm } from "redux-form"
import { useDispatch } from "react-redux"
import { UserType } from "./types"

import { login } from "./redux/actions"
import { Stack, Container, Button } from "@mui/material"

import renderTextField from "../../../../components/fields/inputField"
import renderCheckboxField from "../../../../components/fields/checkboxField"

// import Navbar from "pages/Common/Navbar/index"

import useStyles from "./styles"

const Login = ({ handleSubmit }) => {
  const dispatch = useDispatch()

  const classes = useStyles()

  return (
    <Container>
      <div className={classes.loginContainer}>
        <Stack>
          <Form
            onSubmit={handleSubmit((values: UserType) => dispatch(login(values)))}
          >
            <Field
              name="email"
              color="primary"
              label="Email"
              type="email"
              component={renderTextField}
            />
            <Field
              name="password"
              color="primary"
              label="Password"
              type="password"
              component={renderTextField}
            />
            <Field
              name="rememberMe"
              type="checkbox"
              label="Remember Me"
              color="secondary"
              component={renderCheckboxField}
            />
            <Button 
              type="submit"
            >
    Sign In
            </Button>
          </Form>
        </Stack>
      </div>
    </Container>
  )
}

export default reduxForm({
  form: "LoginForm",
})(Login)
