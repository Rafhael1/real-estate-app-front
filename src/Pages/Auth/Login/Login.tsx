import React from 'react';
import { Form, Field, reduxForm } from 'redux-form';
import { Stack, Box, Modal, Button } from '@mui/material';
import { UserType } from './Login.types';
import { useDispatch } from 'Redux/Hooks';
import { login } from './Login.actions';

import renderTextField from '../../../Components/Fields/InputField';
import renderCheckboxField from '../../../Components/Fields/CheckboxField';

import useStyles from './Login.styles';

const Login = ({ handleSubmit }: any) => {
  const dispatch = useDispatch();

  const styles = useStyles();

  return (
    <Modal open={true} className={styles.boxModal}>
      <Box>
        <Form
          onSubmit={handleSubmit((values: UserType) => dispatch(login(values)))}
        >
          <Stack direction="column" alignItems="center" spacing={0}>
            <Field
              name="email"
              color="primary"
              label="Email"
              type="email"
              autoFocus
              component={renderTextField}
            />
            <Field
              name="password"
              color="primary"
              label="Password"
              type="password"
              component={renderTextField}
            />
            <Field
              name="rememberMe"
              type="checkbox"
              label="Remember Me"
              color="secondary"
              component={renderCheckboxField}
            />
            <Button type="submit">Login</Button>
          </Stack>
        </Form>
      </Box>
    </Modal>
  );
};

export default reduxForm({
  form: 'LoginForm'
})(Login);
