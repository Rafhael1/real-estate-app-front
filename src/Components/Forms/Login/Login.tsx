import React from 'react';
import { Form, Field, reduxForm } from 'redux-form';
import {
  Stack,
  Dialog,
  DialogTitle,
  DialogContent,
  Typography
} from '@mui/material';
import LoadingButton from '@mui/lab/LoadingButton';
import { useDispatch, useSelector } from 'Hooks/Redux';
import { login } from 'Services/Auth/Auth.actions';
import { LoginProps } from './Login.types';
import { UserType } from 'Types/Auth/Auth.types';

import renderTextField from '../../Fields/InputField';
import renderCheckboxField from '../../Fields/CheckboxField';

import useStyles from './Login.styles';

const Login = ({ handleSubmit, isModalOpen, handleModalOpen }: LoginProps) => {
  const dispatch = useDispatch();
  const authState = useSelector((state) => state.Auth);

  const styles = useStyles();

  return (
    <Dialog
      open={isModalOpen}
      onClose={() => handleModalOpen('login')}
      transitionDuration={1}
      PaperProps={{ sx: { borderRadius: '25px', height: '450px' } }}
    >
      <DialogTitle sx={{ alignSelf: 'center' }}>Login</DialogTitle>
      <DialogContent>
        <Form
          onSubmit={handleSubmit(async (values: UserType) => {
            await dispatch(login(values));
            handleModalOpen('login');
          })}
        >
          <Stack direction="column" alignItems="center" spacing={0}>
            <Field
              name="email"
              color="primary"
              label="Email"
              type="email"
              autoFocus
              component={renderTextField}
            />
            <Field
              name="password"
              color="primary"
              label="Password"
              type="password"
              component={renderTextField}
            />
            <Field
              name="rememberMe"
              type="checkbox"
              label="Remember Me"
              color="secondary"
              component={renderCheckboxField}
            />
            <LoadingButton
              loading={authState.isLoading}
              className={styles.loginButton}
              type="submit"
            >
              Login
            </LoadingButton>
            <Typography
              onClick={() => {
                handleModalOpen('register');
              }}
              variant="body2"
              color="textSecondary"
              className={styles.signUpLink}
            >
              {"Don't have an account yet? "}
              <b>Sign Up</b>
            </Typography>
          </Stack>
        </Form>
      </DialogContent>
    </Dialog>
  );
};

export default reduxForm<{}, LoginProps>({
  form: 'LoginForm'
})(Login);
