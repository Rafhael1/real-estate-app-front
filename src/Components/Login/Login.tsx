import React from 'react';
import { useForm } from 'react-hook-form';
import { Stack, Dialog, DialogTitle, DialogContent } from '@mui/material';
import LoadingButton from '@mui/lab/LoadingButton';
import { SignUpLink } from './Login.styles';
import { TextField, CheckboxField } from 'Components';
import { useDispatch, useSelector } from 'Utils/Hooks/Redux';
import { login } from 'Services/Auth/Auth.actions';
import { UserType, AuthProps } from 'Types/Auth/Auth.types';
import useMediaQuery from 'Utils/Hooks/useMediaQuery';
import { useNavigate } from 'react-router';

type UserLogin = Pick<UserType, 'email' | 'password' | 'rememberMe'>;

const Login = ({ isModalOpen, handleModalOpen }: AuthProps) => {
	const dispatch = useDispatch();
	const navigate = useNavigate();
	const authState = useSelector((state) => state.Auth);
	
	const { control, handleSubmit } = useForm({
		defaultValues: {
			email: '',
			password: '',
			rememberMe: true
		}
	});
	const { isMobile } = useMediaQuery();

	const handleOnSubmit = handleSubmit(async(data: UserLogin) => {
		await dispatch(login(data));
		handleModalOpen('login');
		navigate('/', { replace: true })
		window.location.reload()
	});

	return (
		<Dialog
			open={isModalOpen}
			onClose={() => handleModalOpen('login')}
			transitionDuration={1}
			PaperProps={{ sx: { borderRadius: '25px', height: '450px' } }}
		>
			<DialogTitle sx={{ alignSelf: 'center', fontStyle: 'bold' }}>
				Login
			</DialogTitle>
			<DialogContent>
				<form onSubmit={handleOnSubmit}>
					<Stack direction="column" alignItems="center" spacing={0}>
						<TextField
							required
							name="email"
							color="primary"
							label="Email"
							type="email"
							autoFocus
							control={control}
							sx={{ width: isMobile ? '250px' : null }}
						/>
						<TextField
							required
							name="password"
							color="primary"
							label="Password"
							type="password"
							control={control}
							sx={{ width: isMobile ? '250px' : null }}
						/>
						<CheckboxField
							name="rememberMe"
							type="checkbox"
							label="Remember Me"
							color="secondary"
							control={control}
						/>
						<LoadingButton
							loading={authState.isLoading}
							sx={{ width: '200px' }}
							type="submit"
						>
							Login
						</LoadingButton>
						<SignUpLink
							onClick={() => {
								handleModalOpen('register');
							}}
							variant="body1"
							color="textSecondary"
						>
							{"Don't have an account yet? "}
							<b>Sign Up</b>
						</SignUpLink>
					</Stack>
				</form>
			</DialogContent>
		</Dialog>
	);
};

export default Login;
