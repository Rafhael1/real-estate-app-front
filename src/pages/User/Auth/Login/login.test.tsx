/* eslint-disable @typescript-eslint/no-empty-function */
import React from 'react'
import { shallow } from 'enzyme'
import Login from '.'

const funcMock = jest.fn()

const props: any = {
  handleSubmit: funcMock,
}

it('Expect', () => {
  expect(shallow(<Login {...props} />)).toMatchSnapshot()
})