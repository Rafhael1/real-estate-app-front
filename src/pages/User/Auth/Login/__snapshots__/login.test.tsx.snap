// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect 1`] = `
<Hoc
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="LoginForm"
  forwardedRef={null}
  getFormState={[Function]}
  handleSubmit={[MockFunction]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  submitAsSideEffect={false}
  touchOnBlur={true}
  touchOnChange={false}
  updateUnregisteredFields={false}
/>
`;
